### PR TITLE
Add response logging to probe service

### PR DESCRIPTION
**Pull request by dhaus67**
## Description

Add logging of the HTTP response body in case of errors to the probe service.

Previously, if an error occurred, we only saw the HTTP status code within the log message:
```log
5536 probe.go:195] central instance id-42 not reachable: : 500
```

While this may hint to a problem, we can also add the HTTP response body and the error code + reason to the log statement, thus making triaging issues a bit easier.

## Checklist (Definition of Done)

- [ ] Unit and integration tests added
- [ ] Added test description under `Test manual`
- [ ] Documentation added if necessary (i.e. changes to dev setup, test execution, ...)
- [ ] CI and all relevant tests are passing
- [ ] Add the ticket number to the PR title if available, i.e. `ROX-12345: ...`
- [ ] Discussed security and business related topics privately. Will move any security and business related topics that arise to private communication channel.
- [ ] Add secret to app-interface Vault or Secrets Manager if necessary

## Test manual

**TODO:** Add manual testing efforts

```
# To run tests locally run:
make db/teardown db/setup db/migrate
make ocm/setup OCM_OFFLINE_TOKEN=<ocm-offline-token> OCM_ENV=development
make verify lint binary test test/integration
```

### CODE REVIEW

**Comment by dhaus67** [comment]
Current dependencies on/for this PR:
* main
  * **PR #973** <a href="https://app.graphite.dev/github/pr/stackrox/acs-fleet-manager/973" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/stackrox/acs-fleet-manager/973?utm_source=stack-comment).

File: probe/pkg/probe/probe.go [hunk 1]
```diff
@@ -5,11 +5,14 @@ import (
...
 	"crypto/rand"
 	"encoding/hex"
+	"encoding/json"
 	"fmt"
 	"net/http"
 	"strings"
 	"time"
 
+	"github.com/stackrox/rox/pkg/utils"
+
```
Comment (stehessel) [inline probe/pkg/probe/probe.go:15]: Is the newline intentional? Below there is already a `stackrox/rox` module.

File: probe/pkg/probe/probe.go [hunk 2]
```diff
@@ -288,3 +294,16 @@ func retryUntilSucceededWithResponse(ctx context.Context, fn func(context.Contex
 		}
 	}
 }
+
+func extractCentralError(resp *http.Response) string {
+	var centralError public.Error
+	if resp == nil || resp.Body == nil {
+		return ""
+	}
+	defer utils.IgnoreError(resp.Body.Close)
```
Comment (stehessel) [inline probe/pkg/probe/probe.go:303]: Why did you move the `Close` to this function? I would expect the close to be within the same function that declared the response. Otherwise it feels like a side effect of the extract function, even though in this case it doesn't really matter.

File: probe/pkg/probe/probe.go (hunk 1 above)
Comment (dhaus67) [reply probe/pkg/probe/probe.go:15]: Good old golangci-lint

File: probe/pkg/probe/probe.go (hunk 2 above)
Comment (dhaus67) [reply probe/pkg/probe/probe.go:303]: Initially did it since the response body wasn't really used by any other function, but yeah let's add it to the same function where it's declared.

File: probe/pkg/probe/probe.go (hunk 1 above)
Comment (stehessel) [reply probe/pkg/probe/probe.go:15]: What I mean is, if you remove the newline it will be sorted below with the other modules.

Comment (dhaus67) [reply probe/pkg/probe/probe.go]: I thought I fixed it in the last commit, my bad.

**Review by stehessel — APPROVED** [review]
